lib/layout/key.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Layout.Key do
+  @moduledoc """
+  Describes a physical key and its location.
+  """
+
+  @type id :: atom
+
+  @type t :: %__MODULE__{
+          id: id,
+          x: float,
+          y: float,
+          width: float,
+          height: float,
+          led: atom
+        }
+  defstruct [:id, :x, :y, :width, :height, :led]
+
+  def new(id, x, y, opts \\ []) do
+    struct!(__MODULE__,
```

Comment:
:warning: (recommended)

Why not `%__MODULE__{}` here instead of `struct!`?

>when building a struct, as in `struct!(SomeStruct, key: :value)`, it is equivalent to `%SomeStruct{key: :value}` and therefore this function will check if every given key-value belongs to the struct.
>[Kernel.struct!/2](https://hexdocs.pm/elixir/Kernel.html#struct!/2)